x/rollapp/keeper/registered_denoms.go

```diff
@@ -15,6 +18,19 @@ func (k Keeper) HasRegisteredDenom(ctx sdk.Context, rollappID, denom string) (bo
 	return k.registeredRollappDenoms.Has(ctx, collections.Join(rollappID, denom))
 }
 
+func (k Keeper) GetAllRegisteredDenomsPaginated(ctx sdk.Context, rollappID string, pageReq *query.PageRequest) ([]string, *query.PageResponse, error) {
+	denoms, pageRes, err := collcompat.CollectionPaginate(ctx, k.registeredRollappDenoms, pageReq,
+		func(key collections.Pair[string, string], _ collections.NoValue) (string, error) {
+			return key.K1(), nil
+		}, collcompat.WithCollectionPaginationPairPrefix[string, string](rollappID),
+	)
+	if err != nil {
+		return nil, nil, fmt.Errorf("paginate registered denoms: %w", err)
+	}
+
+	return denoms, pageRes, nil
+}
```

Comment:
i think you can skip the error check here
```suggestion
func (k Keeper) GetAllRegisteredDenomsPaginated(ctx sdk.Context, rollappID string, pageReq *query.PageRequest) ([]string, *query.PageResponse, error) {
	return collcompat.CollectionPaginate(ctx, k.registeredRollappDenoms, pageReq,
		func(key collections.Pair[string, string], _ collections.NoValue) (string, error) {
			return key.K1(), nil
		}, collcompat.WithCollectionPaginationPairPrefix[string, string](rollappID),
	)
}
```

Reply:
why didn't I think of that